cfme/generic_objects/instance/rest.py

```diff
@@ -44,6 +45,39 @@ def create(self, name, definition, attributes=None, associations=None):
     return entity
 
 
+@MiqImplementationContext.external_for(GenericObjectInstanceCollection.all, ViaREST)
+def all(self):
+    # get the REST objects
+    rest_generic_objects = self.appliance.rest_api.collections.generic_objects.all
+    if not rest_generic_objects:
+        return []
+    # get the definitions for each generic object
+    def_ids = [gobj.generic_object_definition_id for gobj in rest_generic_objects]
+    # now get the rest gobj class/definition objects
+    rest_generic_object_definitions = [
+        self.appliance.rest_api.collections.generic_object_definitions.find_by(id=id)[0]
+        for id in def_ids
+    ]
+    # now instantiate generic_object_definitions
+    generic_object_definitions = [
+        self.appliance.collections.generic_object_definitions.instantiate(
```

Comment:
**Suggestion:**
Please avoid List-comprehension when logic goes beyond 3 lines.

Reply:
Talked offline, I feel this is more readable than a `for` loop, though it is lengthier than 3 lines.